core/message-handling.go

```diff
@@ -34,66 +36,117 @@ import (
+// incomingMessageHandler using id as the current replica ID and the
+// supplied interfaces.
+func defaultIncomingMessageHandler(id uint32, log messagelog.MessageLog, config api.Configer, stack Stack, logger *logging.Logger) incomingMessageHandler {
 	n := config.N()
+	f := config.F()
+
+	view := func() uint64 { return 0 } // view change is not implemented
+
+	verifyMessageSignature := makeMessageSignatureVerifier(stack)
+	signMessage := makeReplicaMessageSigner(stack)
+	verifyUI := makeUIVerifier(stack)
+	assignUI := makeUIAssigner(stack)
 
 	clientStates := clientstate.NewProvider()
 	peerStates := peerstate.NewProvider()
 
-	view := func() uint64 { return 0 } // view change is not implemented
-	prepareRequestSeq := makeRequestSeqPreparer(clientStates)
-	verifyUI := makeUIVerifier(stack)
+	captureSeq := makeRequestSeqCapturer(clientStates)
+	releaseSeq := makeRequestSeqReleaser(clientStates)
+	prepareSeq := makeRequestSeqPreparer(clientStates)
+	retireSeq := makeRequestSeqRetirer(clientStates)
 	captureUI := makeUICapturer(peerStates)
-	releaseUI := makeUIReleaser(peerStates)
-	collectCommit := defaultCommitCollector(id, clientStates, config, stack)
-	handleGeneratedUIMessage := defaultGeneratedUIMessageHandler(stack, log)
 
-	handleRequest := defaultRequestHandler(id, n, view, stack, clientStates, handleGeneratedUIMessage)
+	handleGeneratedMessage := makeGeneratedMessageHandler(log, clientStates, logger)
+
+	countCommits := makeCommitCounter(f)
+	executeOperation := makeOperationExecutor(stack)
+	executeRequest := makeRequestExecutor(id, executeOperation, signMessage, handleGeneratedMessage)
+	collectCommit := makeCommitCollector(countCommits, retireSeq, executeRequest)
+
+	handleGeneratedUIMessage := makeGeneratedUIMessageHandler(assignUI, handleGeneratedMessage)
+
+	validateRequest := makeRequestValidator(verifyMessageSignature)
+	validatePrepare := makePrepareValidator(n, verifyUI, validateRequest)
+	validateCommit := makeCommitValidator(verifyUI, validatePrepare)
+
+	processRequest := makeRequestProcessor(id, n, view, captureSeq, releaseSeq, prepareSeq, handleGeneratedUIMessage)
+	processPrepare := makePrepareProcessor(id, view, captureUI, prepareSeq, processRequest, collectCommit, handleGeneratedUIMessage)
+	processCommit := makeCommitProcessor(id, view, captureUI, processPrepare, collectCommit)
+
 	replyRequest := makeRequestReplier(clientStates)
-	handlePrepare := makePrepareHandler(id, n, view, verifyUI, captureUI, prepareRequestSeq, handleRequest, collectCommit, handleGeneratedUIMessage, releaseUI)
-	handleCommit := makeCommitHandler(id, n, view, verifyUI, captureUI, handlePrepare, collectCommit, releaseUI)
 
-	return makeMessageHandler(handleRequest, replyRequest, handlePrepare, handleCommit)
-}
+	validateMessage := makeMessageValidator(validateRequest, validatePrepare, validateCommit)
+	processMessage := makeMessageProcessor(processRequest, processPrepare, processCommit)
+	replyMessage := makeMessageReplier(replyRequest)
```

Comment:
From readability viewpoint, the lines of `replyRequest` and `replyMessage` are nice to be adjacent. And maybe `validate*` and `process*` may be ordered in grouped manner, respectively.